lib/ansible/module_utils/common/text/converters.py

```diff
@@ -25,8 +26,59 @@
                                       'surrogate_or_strict',
                                       'surrogate_then_replace'))
 
+_ErrorHandlers = _t.Literal[
+    # Composed error handlers, see above:
+    "surrogate_or_replace",
+    "surrogate_or_strict",
+    "surrogate_then_replace",
+    # Error handlers from https://docs.python.org/3/library/codecs.html#codec-base-classes:
+    "strict",
+    "ignore",
+    "replace",
+    "backslashreplace",
+    "surrogateescape",
+    "xmlcharrefreplace",
+    "namereplace",
+    "surrogatepass",
+]
 
-def to_bytes(obj, encoding='utf-8', errors=None, nonstring='simplerepr'):
+_NonString = _t.Literal[
+    "simplerepr",
+    "empty",
+    "passthru",
+    "strict",
+]
+
+_NonStringNotPassthru = _t.Literal[
+    "simplerepr",
+    "empty",
+    "strict",
+]
```

Comment:
Can these be turned into type aliases?
```suggestion
_ErrorHandlers: _t.TypeAlias = _t.Literal[
    # Composed error handlers, see above:
    "surrogate_or_replace",
    "surrogate_or_strict",
    "surrogate_then_replace",
    # Error handlers from https://docs.python.org/3/library/codecs.html#codec-base-classes:
    "strict",
    "ignore",
    "replace",
    "backslashreplace",
    "surrogateescape",
    "xmlcharrefreplace",
    "namereplace",
    "surrogatepass",
]

_NonString: _t.TypeAlias = _t.Literal[
    "simplerepr",
    "empty",
    "passthru",
    "strict",
]

_NonStringNotPassthru: _t.TypeAlias = _t.Literal[
    "simplerepr",
    "empty",
    "strict",
]
```

Reply:
`TypeAlias` has been added in Python 3.10, so it won't work with Python 3.9.

(I don't really understand why type checking needs to work with every single Python version supported on the target anyway. After all there's `from __future__ import annotations`...)

Reply:
It needs to work because people would be writing target code against those Python versions :man_shrugging:

Reply:
Get it from `typing-extensions`, then: https://typing-extensions.readthedocs.io/en/latest/#typing_extensions.TypeAlias. Perhaps, it's even better to consider `NewType` instead.